pages/login/LoginHandler.inc.php

```diff
@@ -166,104 +170,148 @@ function requestResetPassword($args, $request) {
 		$templateMgr = TemplateManager::getManager($request);
 
 		$email = $request->getUserVar('email');
-		$userDao = DAORegistry::getDAO('UserDAO'); /* @var $userDao UserDAO */
+		$userDao = DAORegistry::getDAO('UserDAO'); /** @var UserDAO $userDao */
 		$user = $userDao->getUserByEmail($email);
 
 		if ($user == null || ($hash = Validation::generatePasswordResetHash($user->getId())) == false) {
-			$templateMgr->assign('error', 'user.login.lostPassword.invalidUser');
-			$templateMgr->display('frontend/pages/userLostPassword.tpl');
+			$templateMgr
+				->assign('error', 'user.login.lostPassword.invalidUser')
+				->display('frontend/pages/userLostPassword.tpl');
 
-		} else {
-			// Send email confirming password reset
-			import('lib.pkp.classes.mail.MailTemplate');
-			$mail = new MailTemplate('PASSWORD_RESET_CONFIRM');
-			$site = $request->getSite();
-			$this->_setMailFrom($request, $mail, $site);
-			$mail->assignParams([
-				'url' => $request->url(null, 'login', 'resetPassword', $user->getUsername(), array('confirm' => $hash)),
-				'siteTitle' => htmlspecialchars($site->getLocalizedTitle()),
-			]);
-			$mail->addRecipient($user->getEmail(), $user->getFullName());
-			$mail->send();
+			return;
+		}
 
-			$templateMgr->assign(array(
-				'pageTitle' => 'user.login.resetPassword',
-				'message' => 'user.login.lostPassword.confirmationSent',
-				'backLink' => $request->url(null, $request->getRequestedPage()),
-				'backLinkLabel' => 'user.login',
-			));
-			$templateMgr->display('frontend/pages/message.tpl');
+		if ($user->getDisabled()) {
+			$templateMgr
+				->assign([
+					'error' => 'user.login.lostPassword.confirmationSentFailedWithReason',
+					'reason' => empty($reason = $user->getDisabledReason() ?? '')
+						? __('user.login.accountDisabled')
+						: __('user.login.accountDisabledWithReason', ['reason' => $reason])
+				])
+				->display('frontend/pages/userLostPassword.tpl');
+			
+			return;
 		}
+
+		// Send email confirming password reset as all check has passed
+		import('lib.pkp.classes.mail.MailTemplate');
+		$mail = new MailTemplate('PASSWORD_RESET_CONFIRM');
+		$site = $request->getSite();
+		$this->_setMailFrom($request, $mail, $site);
+		$mail->assignParams([
+			'url' => $request->url(null, 'login', 'resetPassword', $user->getUsername(), array('confirm' => $hash)),
+			'siteTitle' => htmlspecialchars($site->getLocalizedTitle()),
+		]);
+		$mail->addRecipient($user->getEmail(), $user->getFullName());
+		$mail->send();
+
+		$templateMgr->assign([
+			'pageTitle' => 'user.login.resetPassword',
```

Comment:
[This comment](https://github.com/pkp/pkp-lib/pull/8450/files#r1071637788) should have been on this line of code.

Reply:
Done .